src/main/webapp/app/shared/metis/postings-markdown-editor/postings-markdown-editor.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, Input, OnInit, forwardRef } from '@angular/core';
+import { Command } from 'app/shared/markdown-editor/commands/command';
+import { BoldCommand } from 'app/shared/markdown-editor/commands/bold.command';
+import { ItalicCommand } from 'app/shared/markdown-editor/commands/italic.command';
+import { ReferenceCommand } from 'app/shared/markdown-editor/commands/reference.command';
+import { UnderlineCommand } from 'app/shared/markdown-editor/commands/underline.command';
+import { CodeBlockCommand } from 'app/shared/markdown-editor/commands/codeblock.command';
+import { CodeCommand } from 'app/shared/markdown-editor/commands/code.command';
+import { LinkCommand } from 'app/shared/markdown-editor/commands/link.command';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+    selector: 'jhi-postings-markdown-editor',
+    templateUrl: './postings-markdown-editor.component.html',
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => PostingsMarkdownEditorComponent),
+            multi: true,
+        },
+    ],
+})
+export class PostingsMarkdownEditorComponent implements OnInit, ControlValueAccessor {
+    @Input() content?: string;
+    @Input() maxContentLength: number;
+    defaultCommands: Command[];
+    propagateChange: any = () => {};
+
+    constructor() {}
+
+    /**
+     * on initialization: sets commands that will be available as formatting buttons during creation/editing of postings
+     */
+    ngOnInit(): void {
+        this.defaultCommands = [
+            new BoldCommand(),
+            new ItalicCommand(),
+            new UnderlineCommand(),
+            new ReferenceCommand(),
+            new CodeCommand(),
+            new CodeBlockCommand(),
+            new LinkCommand(),
+        ];
+    }
+
+    /**
+     * writes the current value of a form group and propagates the change
+     * @param value
+     */
+    writeValue(value: string): void {
+        if (value !== undefined) {
```

Comment:
Does string type also take undefined value? Otherwise this check redundant?

Reply:
afaik in typescript any type can take undefined as a value. anyways, I changed this to `value: any` (as [exemplified here](https://angular.io/api/forms/ControlValueAccessor)). The reason why we need to check for `undefined` is that `@Input() content?: string` can be undefined on initialization of the component. Then, this initial value might end up as a parameter to `writeValue()`; hence, the check.